src/frontends/pytorch/include/openvino/frontend/pytorch/decoder.hpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+
+#include "openvino/core/any.hpp"
+#include "openvino/core/node.hpp"
+#include "openvino/core/node_output.hpp"
+#include "openvino/core/partial_shape.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+
+// Extendable type system which reflects TorchScript supported python data types
+// Type nestings are built with the help of ov::Any
+namespace type {
+
+struct Tensor {
+    Tensor() = default;
+    explicit Tensor(const Any& _element_type) : element_type(_element_type) {}
+    Any element_type;
+};
+
+struct Tuple;
+
+struct List {
+    List() = default;
+
+    // Specifies list of elements of element_type type, all elements have the same given type
+    explicit List(const Any& _element_type) : element_type(_element_type) {}
+    Any element_type;
+};
+
+struct Str {};
+
+struct Optional;
+struct Dict;
+struct NamedTuple;
+struct Union;
+
+}  // namespace type
+
+/// Plays a role of node, block and module decoder (kind of temporary fat API)
+struct Decoder {  // TODO: Is it required to be enable_shared_from_this?
+public:
+    // Do not search for input in tensor map; try to access it as a constant of specified type T and return its value
+    // Using Any here is an easy way to avoid template definition, returned object is supposed to be of one of the
+    // fundamental types like int, float etc.
+    virtual Any const_input(size_t index) const = 0;
+
+    // Using size_t for input/output unuque ids are in sync with torch code, see def in
+    // torch/include/torch/csrc/jit/ir/ir.h, Value::unique_
+
+    // TODO: set of input and output methods are not aligned; also they are not aligned with the rest of FEs
+
+    // Input tensor id
+    virtual size_t input(size_t index) const = 0;
+
+    virtual std::vector<size_t> inputs() const = 0;
+
+    // ------------------------------
+    // TODO: physically inputs and outputs refer to PT Values so shape/type is not a property of input/output
```

Comment:
I think we should avoid any `TODO` in such header file. Better to clarify each method.

Reply:
PT FE is not finalized. Any number of TODOs is permissible, I suppose.